src/Compilers/CSharp/Portable/Syntax/DirectiveTriviaSyntax.cs

```diff
@@ -83,7 +83,7 @@ public SyntaxToken DirectiveNameToken
                             }
                         }
                     }
-                    else if (tr.UnderlyingNode == this.Green)
+                    else if (tr.UnderlyingNode == this.Green && tr.SpanStart == this.SpanStart)
```

Comment:
If this should be checking something else I'm happy to change the PR; I'm not sure if there's another gotcha here.

Reply:
Updated with the suggestion of @AlekseyTs.